Fix card.ts - isThrowOut signature

sjcoope opened the pull request:
I've been using your package (which is awesome by the way) to work on a basic Ionic app and have come across an issue.  

When calling `config.isThrowOut` on line 296 of card.ts is uses the incorrect signature and so it doesn't call my override of `config.throwOutConfidence` in my StackConfig. 

I believe I've fixed the issue in this pull request. 

PS: I'm a bit rusty at all this so please forgive me if I've not followed the right process for reporting the issue or raising the pull request.

Thanks!

peterpeterparker commented:
@sjcoope thx for the PR

I'm agree something weird but don't you think it's rather the declaration of isThrowOut in swing.ts (https://github.com/fluster/ionic-swing/blob/1010e38c4f7608d9b3bda93225b5d318a503a070/src/app/modules/ionic-swing/interfaces/swing.ts#L95)  which should be fix respectively replacing `offset` with `coordinateX` and `coordinateY`

swing.js (https://github.com/gajus/swing/blob/9a19ea2fdbf213bb3779b93497088289b355c4cd/src/Card.js#L140) effectively takes four arguments that's why I'm spontaneously thinking that it isn't the Card.ts which would need to be modified but the other declaration

What do you think about it? Could you give a try? If it works, would you like to modify your PR or provide a new one?

Thx in advance for your help

sjcoope committed:
Updated StackConfig Interface

Updated StackConfig.IsThrowOut signature to match signature of function call from card.ts.

sjcoope commented:
Thanks for the reponse @peterpeterparker.  

Now that I've looked over [swing.js](https://github.com/gajus/swing/blob/9a19ea2fdbf213bb3779b93497088289b355c4cd/src/Card.js#L140) and [angular-swing](https://github.com/gajus/angular-swing) I can see you're right, will update the PR now.

Though doesn't this also mean that there's an issue with the [angular-swing](https://github.com/gajus/angular-swing) docs as that provides the wrong example signature for the `isThrowOut` function?  Using the example signature means the `throwOutConfidence` parameter will equal the `HtmlElement` object passed in the [swing js function call](https://github.com/gajus/swing/blob/9a19ea2fdbf213bb3779b93497088289b355c4cd/src/Card.js#L140) and so the check for `throwOutConfidence === 1` will always be false.

peterpeterparker commented:
@sjcoope thx a lot for the updated PR 👍 I have merged it and published a new version v4.1.1 to npm. If you could give it a try and confirm me that's ok that would be cool

regarding the doc, yes agree, I understand it like you, it probably means too that the example isn't correct

sjcoope commented:
Brilliant, thanks.  I've pulled the new version from NPM and tested.  All looks good 👍.

Thanks for your help!

peterpeterparker commented:
@sjcoope awesome, thx for the confirmation and PR 👍
